### PR TITLE
fix: renaming FileScanTask.data_file to data_manifest_entry

### PR DESCRIPTION
**Pull request by a-agmon**
resolves: #299 

This PR is: 

1. Renames FileScanTask.data_file to data_manifest_entry (it returns `ManifestEntry`)
2. Exposes data_file() on `ManifestEntry`
3. change access to data_file to `FileScanTask.data_manifest_entry().data_file()`

### CODE REVIEW

File: crates/iceberg/src/scan.rs [hunk 1]
```diff
@@ -209,8 +209,8 @@ pub struct FileScanTask {
 pub type ArrowRecordBatchStream = BoxStream<'static, crate::Result<RecordBatch>>;
 
 impl FileScanTask {
-    pub fn data_file(&self) -> ManifestEntryRef {
-        self.data_file.clone()
+    pub fn data_manifest_entry(&self) -> ManifestEntryRef {
```
Comment (liurenjie1024) [inline crates/iceberg/src/scan.rs:212]:
```suggestion
    pub fn data(&self) -> ManifestEntryRef {
```
How about just naming it as `data`? cc @marvinlanhenke What do you think?

**Review by liurenjie1024 — COMMENTED** [review]
Hi, @a-agmon Thanks for this fix. I'm just have a small suggestion to the public method, others LGTM! Let's wait a while to see others' comments.

File: crates/iceberg/src/scan.rs (hunk 1 above)
Comment (marvinlanhenke) [reply crates/iceberg/src/scan.rs:212]: yes - i think this will work and still convey the meaning.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**